kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -436,6 +440,123 @@ private static Map<Long, StructField> fieldsById(StructType schema) {
     return columnIdToField;
   }
 
+  private static void validateNestedIds(
+      SchemaChanges schemaChanges,
+      Map<Long, StructField> currentFieldsById,
+      Map<Long, StructField> updatedFieldsById) {
+    for (Tuple2<StructField, StructField> updatedField : schemaChanges.updatedFields()) {
+      validateNestedIdsExist(updatedField._2);
```

Comment:
why not validate consistency here? just curious

Reply:
We could do it in the same method, I just found it easier to read if I broke it up between existence and consistency checks.